### PR TITLE
Change Option<TType>.Text to `string?`

### PR DESCRIPTION
**Pull request by StevenTCramer**
# Pull Request

## 📖 Description

See linked Discussion for more details.

This PR includes the following changes:

1. **Type Update:**
   - Modified the type from [TType?] to [string?] 

2. **Demos Update:**
   - Updated all relevant demos to reflect the type change.

### 🎫 Issues

#2062 

## 📑 Test Plan

There were no tests effected.  My manual testing was to run the demo and insure it still worked.  I ran dotnet test.

### General

- [x] I have added tests for my changes. (There were no tests other than the demo for this.)
- [x] I have tested my changes. (All existing tests passed)
- [x] I have updated the project documentation to reflect my changes. (None found for this class)
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/main/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [x] I have modified a class used by an existing component
- [x] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

File: examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor
Comment (vnbaaij) [inline examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: Can you remove the `.ToString()` from line 23 (`OptionText=...`). Same goes for the other two examples.

**Review by vnbaaij — CHANGES_REQUESTED** [review]

File: examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor
Comment (StevenTCramer) [reply examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: I only changed the ones required to compile.  I could change them to "1" if you prefer.

Comment (StevenTCramer) [reply examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: Ok will update as follows.
```cs
{ new Option<int> { Value = 1, Text = "1", Disabled = true } },
{ new Option<int> { Value = 2, Text = "2" } },
{ new Option<int> { Value = 3, Text = "3" } }
```

Just a min :)

Comment (vnbaaij) [reply examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: That looks better, but was not what I meant :) 

In the component (on line 23) there is now `OptionText="@(i->i.Text.ToString())"` which looks a bit silly now `Text` is already a string. Same for all three samples.

Comment (StevenTCramer) [reply examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: Ok using string literals instead of ToString now :)

Comment (vnbaaij) [reply examples/Demo/Shared/Pages/List/Combobox/Examples/ComboboxFromListOfOptions.razor:1]: > That looks better, but was not what I meant :)
> 
> In the component (on **line 23**) there is now `OptionText="@(i->i.Text.ToString())"` which looks a bit silly now `Text` is already a string. Same for all three samples.

**Review by vnbaaij — APPROVED** [review]

**Pull request merged**